src/app/sold-products/sold-products.component.spec.ts

```diff
@@ -1,14 +1,24 @@
+import { HttpClientModule } from '@angular/common/http';
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
 import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ProductService } from '../product.service';
 
 import { SoldProductsComponent } from './sold-products.component';
 
 describe('SoldProductsComponent', () => {
   let component: SoldProductsComponent;
   let fixture: ComponentFixture<SoldProductsComponent>;
+  let service: ProductService;
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      declarations: [ SoldProductsComponent ]
+      imports: [
+        HttpClientModule
+      ],
+      declarations: [ SoldProductsComponent ],
+      schemas: [
+        CUSTOM_ELEMENTS_SCHEMA
```

Comment:
Why this schema?

Reply:
This was to prevent error:
`ERROR: 'NG0304: 'app-progress-bar' is not a known element:
1. If 'app-progress-bar' is an Angular component, then verify that it is part of this module.
2. If 'app-progress-bar' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.'`

But it seems that I have misunderstood this and I should simply declare ProgressBarComponent in foreach() block instead

Reply:
Use NO_ERRORS_SCHEMA which is correct for this case

Reply:
Done.